compiler/src/dotty/tools/dotc/config/ScalaSettings.scala

```diff
@@ -108,6 +108,7 @@ trait CommonScalaSettings:
   val javaOutputVersion: Setting[String] = ChoiceSetting("-java-output-version", "version", "Compile code with classes specific to the given version of the Java platform available on the classpath and emit bytecode for this version. Corresponds to -release flag in javac.", ScalaSettings.supportedReleaseVersions, "", aliases = List("-release", "--release"))
   val scalaOutputVersion: Setting[String] = ChoiceSetting("-scala-output-version", "version", "Emit TASTy files that can be consumed by specified version of the compiler. The compilation will fail if for any reason valid TASTy cannot be produced (e.g. the code contains references to some parts of the standard library API that are missing in the older stdlib or uses language features unexpressible in the older version of TASTy format)", ScalaSettings.supportedScalaReleaseVersions, "")
 
+  val configuration: Setting[Boolean] = BooleanSetting("-configuration", "Warn about conflicting and redundant tool options", aliases = List("--configuration"))
```

Comment:
First thought: I'm not sure about claiming `--configuration` for this.  I see how it follows the "convention" of deprecation and feature, but... it seems off.  Secondly I'm thinking we should just introduce this as `--configuration`, not as also `-configuration`.

Can we make this a -Xlint option instead?  If not, then I'm happy to be poked to land this as is, especially given I didn't get back to it sooner.

Reply:
Lint is a good idea. I vaguely remember not liking the option. You know it's been a while when
```
Your branch is based on 'upstream/master', but the upstream is gone.
```